pkgs/development/python-modules/apache-airflow/default.nix

```diff
@@ -174,46 +217,41 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
+  # By default, source code of providers is included but unusable due to missing
+  # transitive dependencies. To enable a provider, add it to extraProviders
+  # above
   INSTALL_PROVIDERS_FROM_SOURCES = "true";
 
   postPatch = ''
     substituteInPlace setup.cfg \
-      --replace "importlib_resources~=1.4" "importlib_resources" \
-      --replace "importlib_metadata~=1.7" "importlib_metadata" \
-      --replace "tenacity~=6.2.0" "tenacity" \
-      --replace "pyjwt<2" "pyjwt" \
-      --replace "flask>=1.1.0, <2.0" "flask" \
-      --replace "flask-login>=0.3, <0.5" "flask-login" \
-      --replace "flask-wtf>=0.14.3, <0.15" "flask-wtf" \
-      --replace "jinja2>=2.10.1, <2.12.0" "jinja2" \
-      --replace "attrs>=20.0, <21.0" "attrs" \
-      --replace "cattrs~=1.1, <1.7.0" "cattrs" \
-      --replace "markupsafe>=1.1.1, <2.0" "markupsafe" \
-      --replace "docutils<0.17" "docutils" \
-      --replace "sqlalchemy>=1.3.18, <1.4" "sqlalchemy" \
-      --replace "sqlalchemy_jsonfield~=1.0" "sqlalchemy-jsonfield" \
-      --replace "werkzeug~=1.0, >=1.0.1" "werkzeug" \
-      --replace "itsdangerous>=1.1.0, <2.0" "itsdangerous" \
-      --replace "python-slugify>=3.0.0,<5.0" "python-slugify" \
-      --replace "colorlog>=4.0.2, <6.0" "colorlog"
-
-    substituteInPlace tests/core/test_core.py \
-      --replace "/bin/bash" "${stdenv.shell}"
+      --replace "colorlog>=4.0.2, <5.0" "colorlog" \
+      --replace "pathspec~=0.9.0" "pathspec"
```

Comment:
Notes:
The colorlog shouldn't (hopefully) affect us.
Not sure why the jailbreak of pathspec is here, we have the correct version included
https://github.com/awakesecurity/nixpkgs/blob/0c3bcf979586566c1b252f653cc366aea6465e6e/pkgs/development/python-modules/pathspec/default.nix#L8

Reply:
This change also seems like a good one to apply upstream. You have a really encyclopedic knowledge of airflow by now. I'm sure upstream would love to have your input!